core/crypto/signed_hdr.c

```diff
@@ -87,3 +90,266 @@ TEE_Result shdr_verify_signature(const struct shdr *shdr)
 		return TEE_ERROR_SECURITY;
 	return TEE_SUCCESS;
 }
+
+static const struct shdr_subkey_attr *
+find_attr(const struct shdr_subkey *subkey, uint32_t id)
+{
+	size_t n = 0;
+
+	for (n = 0; n < subkey->attr_count; n++)
+		if (subkey->attrs[n].id == id)
+			return subkey->attrs + n;
+
+	return NULL;
+}
+
+static TEE_Result load_rsa_key(const struct shdr_subkey *subkey,
+			       struct rsa_public_key **key_pp)
+{
+	const uint8_t *base = (const uint8_t *)subkey;
+	const struct shdr_subkey_attr *pub_exp = NULL;
+	const struct shdr_subkey_attr *modulus = NULL;
+	struct rsa_public_key *key = NULL;
+	TEE_Result res = TEE_SUCCESS;
+
+	pub_exp = find_attr(subkey, TEE_ATTR_RSA_PUBLIC_EXPONENT);
+	if (!pub_exp)
+		return TEE_ERROR_SECURITY;
+	modulus = find_attr(subkey, TEE_ATTR_RSA_MODULUS);
+	if (!modulus)
+		return TEE_ERROR_SECURITY;
+
+	key = calloc(1, sizeof(*key));
+	if (!key)
+		return TEE_ERROR_SECURITY;
```

Comment:
`TEE_ERROR_OUT_OF_MEMORY` would be more helpful, any concern with it?

Reply:
No, I'll change.